src/core/server/saved_objects/service/lib/repository.test.js

```diff
@@ -155,27 +155,39 @@ describe('SavedObjectsRepository', () => {
     log: {},
   });
 
-  const getMockGetResponse = ({ type, id, references, namespace, originId }) => ({
-    // NOTE: Elasticsearch returns more fields (_index, _type) but the SavedObjectsRepository method ignores these
-    found: true,
-    _id: `${registry.isSingleNamespace(type) && namespace ? `${namespace}:` : ''}${type}:${id}`,
-    ...mockVersionProps,
-    _source: {
-      ...(registry.isSingleNamespace(type) && { namespace }),
-      ...(registry.isMultiNamespace(type) && { namespaces: [namespace ?? 'default'] }),
-      ...(originId && { originId }),
-      type,
-      [type]: { title: 'Testing' },
-      references,
-      specialProperty: 'specialValue',
-      ...mockTimestampFields,
-    },
-  });
+  const getMockGetResponse = (
+    { type, id, references, namespace: objectNamespace, originId },
+    namespace
+  ) => {
+    const namespaceId =
+      // eslint-disable-next-line no-nested-ternary
+      objectNamespace !== undefined
+        ? objectNamespace !== 'default'
+          ? objectNamespace
+          : undefined
+        : namespace;
```

Comment:
I think this could be simplified to:

```suggestion
    const namespaceId = objectNamespace === 'default'
      ? undefined
      : objectNamespace ?? namespace;
```